common/src/main/java/org/mcaccess/minecraftaccess/features/point_of_interest/POIBlocks.java

```diff
@@ -179,49 +211,22 @@ private void checkBlock(BlockPos blockPos, int val) {
             return;
         }
 
-        boolean shouldPlayMarkedOnly = onPOIMarkingNow && POIMarkingConfigMap.getInstance().isSuppressOtherWhenEnabled();
-
-        for (POIGroup group : builtInGroups.values()) {
-            if (group.checkAndAddBlock(blockState, blockPos)) {
-                if (playSound && playSoundForOtherBlocks && !shouldPlayMarkedOnly) {
-                    world.playSound(player, blockPos, group.sound, SoundCategory.BLOCKS, volume, group.soundPitch);
-                }
-            }
-        }
-
-        if (playSound && !playSoundForOtherBlocks && !shouldPlayMarkedOnly) {
-            POIGroup oreGroup = builtInGroups.get("ore");
-
-            for (BlockPos pos : oreGroup.getBlocks().keySet()) {
-                world.playSound(player, pos, oreGroup.sound, SoundCategory.BLOCKS, volume, oreGroup.soundPitch);
-            }
-        }
-
-        if (shouldPlayMarkedOnly) {
-            POIGroup markedGroup = builtInGroups.get("marked");
-
-            for (BlockPos pos : markedGroup.getBlocks().keySet()) {
-                world.playSound(player, pos, markedGroup.sound, SoundCategory.BLOCKS, volume, markedGroup.soundPitch);
+        for (POIGroup<BlockPos> group : groups) {
+            if (group.add(blockPos)) {
+                break;
             }
         }
     }
 
-    private void setMarkedBlock(Block block) {
-        this.markedBlock = block == null ? s -> false : s -> s.isOf(block);
+    private void setMarkedBlock(@Nullable Block block) {
+        markedBlock = block;
     }
 
-    public List<TreeMap<Double, Vec3d>> getLockingCandidates() {
-        List<TreeMap<Double, Vec3d>> results = new ArrayList<>();
-
-        if (onPOIMarkingNow && POIMarkingConfigMap.getInstance().isSuppressOtherWhenEnabled()) {
-            results.add(builtInGroups.get("marked").getBlocks(true));
-            return results;
-        }
-
-        for (POIGroup group : builtInGroups.values()) {
-            results.add(group.getBlocks(true));
+    public @UnmodifiableView List<BlockPos> getLockingCandidates() {
+        if (isMarking && POIMarkingConfigMap.getInstance().isSuppressOtherWhenEnabled()) {
+            return markedGroup.getItems();
         }
```

Comment:
Want a little `else` for clarifying the `if-condition`?

Reply:
An else statement would be superfluous here and would typically be regarded as less clear because it makes the code more deeply nested.